### PR TITLE
add Nearfield, fix typo in Le Chiffre Readme

### PR DESCRIPTION
**Pull request by tominabox1**
## Description
Initial commit of the Nearfield keyboard. Dr. Doof is the original creator of the keyboard source code, I've volunteered to maintain it and submit this PR.

Also sneaking in a typo fix on Le Chiffre's readme.md

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cannonkeys/nearfield/readme.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ 
+* Keyboard Maintainer: [TJ Campie](https://github.com/tominabox1)
+* Hardware Supported: Nearfield
+* Hardware Availability: Cannonkeys
+
+Make example for this keyboard (after setting up your build environment):
+
+    make nearfield:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (keyboard-magpie) [inline keyboards/cannonkeys/nearfield/readme.md:23]: Can you provide examples of how to get the board into bootloader, please?

File: keyboards/cannonkeys/nearfield/config.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 3
+#endif
+
+#ifdef RGB_DI_PIN
+#define RGBLED_NUM 0
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/cannonkeys/nearfield/config.h:52]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/cannonkeys/nearfield/readme.md (hunk 1 above)
Comment (drashna) [reply keyboards/cannonkeys/nearfield/readme.md:23]: Could you add instructions on how to enter the bootloader, to flash new firmware?  This is needed, as a new user may not be aware of how to enter to bootloader to flash new firmware.

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by tominabox1 — COMMENTED** [review]
Please pause temporarily, I've been informed that there was an error that needs testing on hardware. I think its resolved but waiting for confirmation from Dr. Doof

**Comment by tominabox1** [comment]
@drashna Ready to move ahead now, thanks!!!

File: keyboards/cannonkeys/nearfield/config.h [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x0004
+#define PRODUCT_ID      0x0002
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    JLC
+#define PRODUCT         nearfield
+
```
Comment (fauxpark) [inline keyboards/cannonkeys/nearfield/config.h:28]:
```suggestion
```

File: keyboards/cannonkeys/nearfield/info.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+    "keyboard_name": "Nearfield",
+    "url": "",
+    "maintainer": "tominabox1",
```
Comment (fauxpark) [inline keyboards/cannonkeys/nearfield/info.json:4]:
```suggestion
    "keyboard_name": "Nearfield",
    "manufacturer": "JLC",
    "url": "",
    "maintainer": "tominabox1",
    "usb": {
        "vid": "0x0004",
        "pid": "0x0002",
        "device_version": "0.0.1",
    },
```

File: keyboards/cannonkeys/nearfield/readme.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# Nearfield
+
+![Nearfield](https://i.imgur.com/hNj44FY.jpg)
```
Comment (fauxpark) [inline keyboards/cannonkeys/nearfield/readme.md:3]:
```suggestion
![Nearfield](https://i.imgur.com/hNj44FYh.jpg)
```

File: keyboards/cannonkeys/nearfield/rules.mk [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   comment out to disable the options.
```
Comment (fauxpark) [inline keyboards/cannonkeys/nearfield/rules.mk:8]:
```suggestion
#   change yes to no to disable
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cannonkeys/nearfield/config.h (hunk 2 above)
Comment (drashna) [reply keyboards/cannonkeys/nearfield/config.h:45]:
```suggestion
```

Comment (tominabox1) [reply keyboards/cannonkeys/nearfield/config.h:45]: ope turns out the board doesnt have RGB at all lol, removed in latest push

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**